cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/app/etl/batch/ETLMapReduce.java

```diff
@@ -279,12 +348,27 @@ public void map(Object key, Object value, Context context) throws IOException, I
         KeyValue<Object, Object> input = new KeyValue<>(key, value);
         TransformResponse<Object> recordsAndErrors = transformExecutor.runOneIteration(input);
         Iterator<Object> transformedRecords = recordsAndErrors.getEmittedRecords();
+        Set<String> transformsWithoutErrorDataset = new HashSet<>();
```

Comment:
this should be in the mapper instead of the map method. Otherwise we can warn multiple times instead of just once.